judge/views/blog.py

```diff
@@ -136,15 +136,24 @@ def get_context_data(self, **kwargs):
 
 class PostList(PostListBase):
     template_name = 'blog/list.html'
+    show_all_blogs = False
+    tab = 'home'
 
     def get_queryset(self):
         queryset = super(PostList, self).get_queryset()
-        queryset = queryset.filter(global_post=True)
+        self.show_all_blogs = self.request.GET.get('show_all_blogs', 'false') == 'true'
+        queryset = queryset.filter(global_post=not self.show_all_blogs)
```

Comment:
this will either show global_post or show non global_post, it will never show all posts.

You should use:
```py
if not self.show_all_blogs:
    query = query.filter(global_post=True)
```